### DISCUSSION TITLE
%RELEASE_TITLE% not written to ALBUM tag

### DISCUSSION THREAD

**Discussion by gitgetgot** [ghDaYuYu/foo_discogs#61] [Spam/Abuse]
Version: 1.0.4
Compiled: Aug 19 2021

The ALBUM Tag is not being written, although, it is the default setting.

My tag mappings for ALBUM:
![2021-08-20 22-53-27_Tag Mapping](https://user-images.githubusercontent.com/974980/130293586-8e9bcb2c-a7e5-4e45-aa0d-8491ecc42a37.png)

I have tagged an album "TEST TEST TEST " and the [album](https://www.discogs.com/release/23963) written should be "The State Of E:Motion Vol. 7", but after writing the tags with the fine component, the album is still "TEST TEST TEST "

Video:
![albumnotwritten](https://user-images.githubusercontent.com/974980/130293581-c6fb4edf-52d2-4455-82c3-c1598a29acb4.gif)

EDIT:
also the TITLE is not written always.
my formatting string (which works flawlessly in the old component):
`$ifequal(%TRACK_TOTAL_HIDDEN_TRACKS%,0,%TRACK_TITLE%,$join($extend($array(%TRACK_TITLE%),%<TRACK_HIDDEN_TRACKS_TITLE>%),' / '))`
![2021-08-21 04-16-18_Tag Mapping](https://user-images.githubusercontent.com/974980/130307504-2054f3b4-9ba5-4a56-9283-ecfe1f792cdf.png)

![aain](https://user-images.githubusercontent.com/974980/130307519-4b186fa8-9643-4ce3-905b-b345f1fe792c.gif)

**Comment by ghuDaYuYu** [comment]
Are you sure you want a write operation, I would say you need an update instead.

Look at your stats:
Title -> 12 skipped, 37 equal
Album -> 49 skipped


Btw, thanks for taking the time and your effort adding all these issues...
It will take quite some time to check them all though : )

Regards

**Reply by gitgetgot** [reply]
> Are you sure you want a write operation, I would say you need an update instead.

Yes, I want a write.

As I understood write & update (from zoomorphs'/bubblegum's ancient posts), writes are written always to the files and updates are only written when initiated through the "Update tags..." dialogue. That is exactly what we need here and do every day :-)

(We only update some tags, say, after a year has passed for a specific release, but most tags are written manually (i.e. date, album) and should only be overwritten by the update-dialogue, when we select "write" instead of "update")

>thanks for taking the time and your effort adding all these issues... It will take quite some time to check them all though 
No worries, its great to see, someone with skills has seriously taken the development and offers time and $brain to it!!
This is very much appreceated. Not only by me!

btw: I almost never use the preview. I just showed it here, because IMHO it would be easier to realize the "bug".
I have no idea that the stats mean and don't care too ;-) The only thing that counts are the tags, we see in the properties dialogue of fb2k :-))

**Comment by frogworth** [comment]
I think that you have these terms wrong.
As I recall, "write" only writes to tags which do not currently have content, while "write + update" means also to write to tags which already have content (i.e. update the content if present).

This means that if you want to always keep your current values for a certain tag, you'll make it only "write" in that dialog.

So try changing ALBUM to "write + update" - it should do what you're looking for.

**Comment by gitgetgot** [comment]
Well,
we used it for years the way we thought it should work :-)

Obviously, there is a difference between the old and the modded component now. 

I have to admit, with your and my explanation [back in 2017](https://hydrogenaud.io/index.php?topic=50523.msg933831#msg933831)  the old component behaves "wrong" and the modded now works as described.
I guess, Users have to change their behaviour/use cases now...

My tests changing ALBUM to "write + update" worked as you told, 
unfortunately, 
I can not do any tests with updating tags, because the menu (Update Tags...) is greyed out and (after a restart of a clean installed fb2k) no dialogue window is comming up here too.

I'll wait some time to continue further tests, until maybe something has changed that greyed out thingy.... (On my ride back in time in the components forum, I have read that zoomorph [had fixed](https://hydrogenaud.io/index.php?topic=50523.msg922619#msg922619) some of my reporded greyed out issues. At least the one in the post, I found...)
